### PR TITLE
Self contained cli+lib

### PR DESCRIPTION
**Pull request by why-not-try-calmer**
_Todo_
- [x] lib to Python 3, typing
- [x] basic cli
- [x] refactoring lib using python-transifex (SDK)
- [ ] add nice features:
  - [ ] get number / percentage of translations relative to src
  - [ ] add special translation strings to translate config-only values
  - [x] async pull & fetch
- [x] enough tests for lib
- [ ] enough tests for cli

### CODE REVIEW

**Comment by why-not-try-calmer** [comment]
(Sorry, I thought I could stop using my fork but in fact I still need to host this PR.)

File: setup.py [hunk 1]
```diff
@@ -1,35 +1,36 @@
-from setuptools import setup
 import sys
 
-python_min_version = (3, 6)
+from setuptools import setup
+
+python_min_version = (3, 10)
```
Comment (3nids) [inline setup.py:5]: do we need such a high requirement?

File: pytransifex/api_old.py [hunk 2]
```diff
@@ -1,17 +1,18 @@
-#/usr/bin/python3
-
-
-import os
 import codecs
-import requests
```
Comment (3nids) [inline pytransifex/api_old.py:6]: shall we drop the old file?

**Review by 3nids — COMMENTED** [review]
can you add the tests on CI?

File: setup.py (hunk 1 above)
Comment (why-not-try-calmer) [reply setup.py:5]: No we don't, but I thought that if Python is exclusively run from Docker, and assuming that 3.10 is backward compatible quite a long way back, we might as well push the day where we need to update that far.

If you nonetheless prefer 3.9 or .8, no problem though.

**Comment by why-not-try-calmer** [comment]
Pursuing on #4 as this PR is stale.

**Pull request closed without merging**